streaming/src/main/scala/org/apache/spark/streaming/util/FileBasedWriteAheadLog.scala

```diff
@@ -313,9 +313,7 @@ private[streaming] object FileBasedWriteAheadLog {
     val taskSupport = new ExecutionContextTaskSupport(executionContext)
     val groupSize = taskSupport.parallelismLevel.max(8)
     source.grouped(groupSize).flatMap { group =>
-      val parallelCollection = group.par
-      parallelCollection.tasksupport = taskSupport
-      parallelCollection.map(handler)
+      ThreadUtils.parmap(group)(handler)(executionContext)
```

Comment:
I would prefer to not change DStream codes.

Reply:
Are there any specific reasons, or you think `parmap` is not reliable enough?